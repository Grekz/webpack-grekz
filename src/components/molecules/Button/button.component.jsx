// @flow
import * as React from 'react'
import type { ButtonType } from '../../../custom-types'
import styled from 'react-emotion'

const DefaultButton = styled('button')`
  box-sizing: border-box;
  font-size: 18px;
  min-height: 40px;
  min-width: 130px;
  padding: 8px 30px;
  border-radius: 50px;
  display: inline-block;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  white-space: nowrap;
  line-height: 1.42857143;
  user-select: none;
  will-change: transform;
  &:hover,
  &:focus {
    box-shadow: none;
    text-decoration: none;
  }
`
const PrimaryButton = styled(DefaultButton)`
  color: #fff;
  background-color: #fb4c4c;
  border-color: #fb4c4c;
  &:hover,
  &:focus {
    color: #fff;
    background-color: #e23232;
    border-color: #e23232;
    outline: 0 !important;
  }
  &:active {
    color: #fff;
    background-color: #e23232;
    border-color: #e23232;
    transform: scale(0.97);
    transition-duration: 300ms;
  }
`
const SecondayButton = styled(DefaultButton)``
const Button = ({ as = 'button', ...props }: ButtonType) => {
  let Element = props.secondary ? SecondayButton : PrimaryButton
  Element = as == 'button' ? Element : Element.withComponent('a')
  return <Element {...props} />
}
export default Button
