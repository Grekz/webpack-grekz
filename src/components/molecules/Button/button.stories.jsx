import React from 'react'

import { storiesOf } from '@storybook/react'
// import { action } from '@storybook/addon-actions'
// import { linkTo } from '@storybook/addon-links'

import Button from './Button'

storiesOf('Button', module)
  .add('Default', () => <Button alt="test">Tests</Button>)
  .add('As Anchor', () => <Button as="a">As Link</Button>)
