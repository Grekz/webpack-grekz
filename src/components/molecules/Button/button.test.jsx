import React from 'react'
import Button from './button.component'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<Button /> rendering', () => {
  it('should render Button', () => {
    const wrapper = mount(<Button />)
    expect(wrapper).toHaveLength(1)
  })
  it('should contain 1 button element', () => {
    const wrapper = mount(<Button />)
    expect(wrapper.find('button')).toHaveLength(1)
  })
  it('should contain 1 anchor element', () => {
    const wrapper = mount(<Button as="a" />)
    expect(wrapper.find('a')).toHaveLength(1)
  })
  it('mathces the snapshot', () => {
    const tree = mount(<Button />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
