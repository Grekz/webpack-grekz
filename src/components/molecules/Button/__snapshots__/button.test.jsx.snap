// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering mathces the snapshot 1`] = `
<Button>
  <Styled(button)>
    <button
      className="css-dwyr59 e1hsd3xl1"
    />
  </Styled(button)>
</Button>
`;
